src/glyph/parse.rs

```diff
@@ -123,18 +123,66 @@ impl<'names> GlifParser<'names> {
             }
         }
 
+        // UFO contour sanity constraints:
+        // 1. If a "move" occurs, it must be the first point of a contour. ufoLib
+        //    may allow this for format 1, we don't.
+        // 2. The "smooth" attribute must not be set on off-curve points.
+        // 3. An off-curve point must be followed by a curve or qcurve.
+        // 4. A maximum of two offcurves can precede a curve.
         let mut points = Vec::new();
+        let mut number_of_offcurves: u32 = 0;
+        let mut contour_closed: bool = true;
```

Comment:
instead of maintaining a flag for this, I would just wait until we have all the points and then do,

```rust
let is_closed = points.first().map(|p| p.typ) == Some(PointType::Move);
```